Model/ProductExporter.php

```diff
@@ -0,0 +1,402 @@
+        'Bottom price',
+        'Bottom price with VAT',
+        'Cost price',
+        'Cost price with VAT',
+        'Product kind alias',
+        'Configurable product sku',
+        'Category',
+        'Category slug',
+        'Extra Category slugs',
+        'Extra Label slugs',
+        'Attribute set name',
+        'Attribute set alias',
+        'Sales active',
+        'Sales relation limited',
+        'Merk (raw)',
+        'Merk (label)',
+        'Color (raw)',
+        'Color (label)',
+        'Kleuren (raw)',
+        'Kleuren (label)',
+        'Maten (raw)',
+        'Maten (label)',
+        'Beschrijving nodig op kassa (raw)',
+        'Beschrijving nodig op kassa (label)',
+        'Heeft gewicht nodig op kassa (raw)',
+        'Heeft gewicht nodig op kassa (label)',
+        'Verkoopeenheid (raw)',
+        'Verkoopeenheid (label)',
+        'Vulpennen (raw)',
+        'Vulpennen (label)',
+        'Barcode (raw)',
+        'Barcode (label)',
+        'Image 1'
+    ];
+    const VAR_EXPORT_STOREKEEPER = 'var_export_storekeeper';
+
+    private CollectionFactory $productCollectionFactory;
+    private Csv $csv;
+    private Filesystem $filesystem;
+    private DirectoryList $directoryList;
+    private File $file;
+    private StoreManagerInterface $storeManager;
+    private StockRegistryInterface $stockRegistry;
+    private TaxCalculationInterface $taxCalculation;
+    private Calculation $calculation;
+    private TaxClassRepositoryInterface $taxClassRepository;
+    private RateCollectionFactory $rateCollectionFactory;
+    private ResourceCalculation $resourceCalculation;
+    private TaxCalculationCollectionFactory $taxCalculationCollectionFactory;
+    private TaxRateRepositoryInterface $taxRateRepository;
+    private ProductHelper $productHelper;
+    private CategoryRepositoryInterface $categoryRepository;
+    private SetFactory $attributeSetFactory;
+    private ImageFactory $imageFactory;
+    private DateTime $dateTime;
+
+    public function __construct(
+        CollectionFactory $productCollectionFactory,
+        Csv $csv,
+        Filesystem $filesystem,
+        DirectoryList $directoryList,
+        File $file,
+        StoreManagerInterface $storeManager,
+        StockRegistryInterface $stockRegistry,
+        TaxCalculationInterface $taxCalculation,
+        Calculation $calculation,
+        TaxClassRepositoryInterface $taxClassRepository,
+        RateCollectionFactory $rateCollectionFactory,
+        ResourceCalculation $resourceCalculation,
+        TaxCalculationCollectionFactory $taxCalculationCollectionFactory,
+        TaxRateRepositoryInterface $taxRateRepository,
+        ProductHelper $productHelper,
+        CategoryRepositoryInterface $categoryRepository,
+        SetFactory $attributeSetFactory,
+        ImageFactory $imageFactory,
+        DateTime $dateTime
+    ) {
+        $this->productCollectionFactory = $productCollectionFactory;
+        $this->csv = $csv;
```

Comment:
The csv should in separate class, this should only yield row per row. See: [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).